packages/code-studio/src/main/SessionUtils.ts

```diff
@@ -64,4 +83,42 @@ export function createCoreClient(): CoreClient {
   return new dh.CoreClient(websocketUrl);
 }
 
+export async function requestParentLoginOptions(): Promise<LoginOptions> {
+  if (window.opener == null) {
+    throw new Error('window.opener is null, unable to send auth request.');
+  }
+  return new Promise(resolve => {
+    const listener = (
+      event: MessageEvent<{
+        message: string;
+        payload: LoginOptions;
+      }>
+    ) => {
+      const { data } = event;
+      log.info('Received message', data);
+      if (data?.message !== 'loginOptions') {
+        log.info('Ignore received message', data);
```

Comment:
We should change these to `debug` level, as it may have login credential information which we don't want in the logs.